From a3c5f6c326450610b9f121ec3e37e3aa19ad5c28 Mon Sep 17 00:00:00 2001
From: Entrpi <entrpi@proton.me>
Date: Thu, 5 Feb 2026 23:21:52 +1100
Subject: [PATCH 055/145] net: realtek: r8125/r8126: Enable WoL and RSS support

Enable additional features for Realtek r8125 and r8126 drivers:
- ENABLE_S0_MAGIC_PACKET: Wake-on-LAN magic packet support
- ENABLE_RSS_SUPPORT: Receive Side Scaling for improved RX performance

These features improve power management and network throughput on
multi-core systems.
---
 drivers/net/ethernet/realtek/r8125/Makefile | 2 ++
 drivers/net/ethernet/realtek/r8126/Makefile | 2 ++
 2 files changed, 4 insertions(+)

diff --git a/drivers/net/ethernet/realtek/r8125/Makefile b/drivers/net/ethernet/realtek/r8125/Makefile
index 1a8b367a6452..a18ac30b1a5e 100644
--- a/drivers/net/ethernet/realtek/r8125/Makefile
+++ b/drivers/net/ethernet/realtek/r8125/Makefile
@@ -5,8 +5,10 @@ r8125-objs := r8125_n.o rtl_eeprom.o rtltool.o
 ccflags-y += -DCONFIG_R8125_NAPI
 ccflags-y += -DCONFIG_R8125_VLAN
 ccflags-y += -DENABLE_S5WOL
+ccflags-y += -DENABLE_S0_MAGIC_PACKET
 ccflags-y += -DENABLE_EEE
 ccflags-y += -DENABLE_TX_NO_CLOSE
 ccflags-y += -DENABLE_GIGA_LITE
 ccflags-y += -DCONFIG_SOC_LAN
 ccflags-y += -DCONFIG_ASPM
+ccflags-y += -DENABLE_RSS_SUPPORT
diff --git a/drivers/net/ethernet/realtek/r8126/Makefile b/drivers/net/ethernet/realtek/r8126/Makefile
index 293c70a7a6bd..8f0ccddb3abf 100644
--- a/drivers/net/ethernet/realtek/r8126/Makefile
+++ b/drivers/net/ethernet/realtek/r8126/Makefile
@@ -5,8 +5,10 @@ r8126-objs := r8126_n.o rtl_eeprom.o rtltool.o
 ccflags-y += -DCONFIG_R8126_NAPI
 ccflags-y += -DCONFIG_R8126_VLAN
 ccflags-y += -DENABLE_S5WOL
+ccflags-y += -DENABLE_S0_MAGIC_PACKET
 ccflags-y += -DENABLE_EEE
 ccflags-y += -DENABLE_TX_NO_CLOSE
 ccflags-y += -DENABLE_GIGA_LITE
 ccflags-y += -DCONFIG_SOC_LAN
 ccflags-y += -DCONFIG_ASPM
+ccflags-y += -DENABLE_RSS_SUPPORT
-- 
2.54.0

